recipes/WOA_1degree_monthly/recipe.py

```diff
@@ -0,0 +1,55 @@
+# Recipe for WOA 1degree data 
+# Data description: https://www.ncei.noaa.gov/sites/default/files/2020-04/woa18documentation.pdf 
+
+from pangeo_forge_recipes import patterns
+from pangeo_forge_recipes.recipes import XarrayZarrRecipe
+
+
+def make_url(variable, time):
+    # Many cases needed for different variables
+    # Here we might not even use all
+    if (variable == 'temperature') or (variable == 'salinity'):
+        fname =("https://www.ncei.noaa.gov/thredds-ocean/fileServer/ncei/woa/"
+            f"{variable}/decav/1.00/woa18_decav_{variable[0]}{time:02d}_01.nc")
+    elif variable == 'density':
+        fname = ("https://www.ncei.noaa.gov/thredds-ocean/fileServer/ncei/woa/"
+            f"{variable}/decav/1.00/woa18_decav_I{time:02d}_01.nc")
+    elif variable == 'oxygen' or variable == 'AOU' or variable == 'phosphate' or variable == 'nitrate':
+        fname = ("https://www.ncei.noaa.gov/thredds-ocean/fileServer/ncei/woa/"
+            f"{variable}/all/1.00/woa18_all_{variable[0]}{time:02d}_01.nc")
+    elif variable == 'silicate':
+        fname = ("https://www.ncei.noaa.gov/thredds-ocean/fileServer/ncei/woa/"
+            f"{variable}/all/1.00/woa18_all_i{time:02d}_01.nc")
+    elif variable == 'mld':
+        fname = ("https://www.ncei.noaa.gov/data/oceans/woa/WOA18/DATA/"
+            f"{variable}/netcdf/decav81B0/1.00/woa18_decav81B0_M02{time:02d}_01.nc")
+
+    return fname
+
+
+variable_merge_dim = patterns.MergeDim("variable", keys=["temperature", 
+                                                         "salinity",
+                                                         "density",
+                                                         "oxygen", 
+                                                         "AOU", 
+                                                         #"phosphate", # these following variables do not have data to 1500m in monthly
+                                                         #"nitrate", 
+                                                         #"silicate", 
+                                                         #"mld"
+                                                        ])
+
+month_concat_dim = patterns.ConcatDim("time", keys=list(range(1, 13)), nitems_per_file=1)
+
+pattern = patterns.FilePattern(format_function, variable_merge_dim, month_concat_dim)
```

Comment:
```suggestion
pattern = patterns.FilePattern(make_url, variable_merge_dim, month_concat_dim)
```

Reply:
@dhruvbalwada, this is the root of the `NameError` which is confounding @pangeo-forge-bot.

Can you accept the suggestion when you get a moment?